### PR TITLE
Alternate teacher

### PR DESCRIPTION
**Pull request by DuToit-Hexagon**
Added an alternate responsible teacher that schools must fill in when registering students. Also split the email field into a work and personal email for the responsible teachers, as well as a cellphone field. Made the surname field compulsory and allows the report_downloaded and answer_sheet_downloaded to be null as otherwise the website broke when trying to send out confirmation emails after registration. Also added a isPrimary field to allow the database to distinguish between primary and alternate responsible teachers.

Updated the school summary sheet to include the alternate teacher fields. Updated reports generator to generate personalised emails to both the primary and alternate responsible teachers.

### CODE REVIEW

File: uctMaths/competition/reports.py [hunk 1]
```diff
@@ -54,13 +55,32 @@ def send_answer_sheets(school, answer_sheet, cc_admin=False):
...
     output_string += UMC_header("Answer Sheets")
     output_string += 'Answer sheets for %s\n%s\n'%(school.name, UMC_datetime())
-    recipient_list = [rteacher.email]
+    recipient_list = [rteacher.email_school]
     if cc_admin:
         recipient_list.append(compadmin.admin_emailaddress())
+    send_email(
+        "(Do not reply) UCT Mathematics Competition %s Answer Sheets %s" % (school.name, recipient_list),
+        output_string,
+        "UCT Mathematics Competition <UCTMathsCompetition@j5int.com>",
```
Comment (ianwj5int) [inline uctMaths/competition/reports.py:64]: This address doesn't exist and j5int.com isn't a domain that we use any more. I think this should change to something like noreply@uctmathscompetition.org.za?
There are other places in the site that also use j5int.com addresses. They should be updated as well.

Comment (yaseen-mowzer-hexagon) [reply uctMaths/competition/reports.py:64]: I don't think we should use a fake email address, as spam filters might not like that. What is the address are we sending emails from?

File: uctMaths/competition/compadmin.py [hunk 2]
```diff
@@ -685,6 +685,9 @@ def assign_student_awards():
                         student.save()
                     else:
                         break
+    
+    student_list = SchoolStudent.objects.all()
```
Comment (yaseen-mowzer-hexagon) [inline uctMaths/competition/compadmin.py:689]: What does this line do?

Comment (DuToit-Hexagon) [reply uctMaths/competition/compadmin.py]: I think that was left over from the earliest tests I did - I thought I removed it before, but it should be gone now

File: uctMaths/competition/reports.py (hunk 1 above)
Comment (yaseen-mowzer-hexagon) [reply uctMaths/competition/reports.py]: So I spoke to Stephen, and it sounds like our old mail server has been taken down and we need to reconfigure this.

I think we should leave this as is for this MR and fix the issues with the mail server in a different MR.

**Comment by DuToit-Hexagon** [comment]
I got the following traceback in the process_tasks window when sending results to a school:

`2022-1-20 8:25:54: Emailing results for school with ID: 17
WARNING:root:Rescheduling Task(uctMaths.background_tasks.bg_email_results)
Traceback (most recent call last):
  File "C:\Users\dspies\work\uct\git\venv\uctmaths_venv\lib\site-packages\background_task\tasks.py", line 49, in bg_runner
    func(*args, **kwargs)
  File "C:\Users\dspies\work\uct\git\uct-maths-competition\uctMaths\uctMaths\background_tasks.py", line 37, in bg_email_results
    send_results(school, result, True)
  File "C:\Users\dspies\work\uct\git\uct-maths-competition\uctMaths\competition\reports.py", line 39, in send_results
    'UCT Mathematics Competition <UCTMathsCompetition@j5int.com>',#from
  File "C:\Users\dspies\work\uct\git\uct-maths-competition\uctMaths\competition\reports.py", line 97, in send_email
    'your students will write on. \n\n' \
  File "C:\Users\dspies\work\uct\git\venv\uctmaths_venv\lib\site-packages\django\core\mail\message.py", line 283, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "C:\Users\dspies\work\uct\git\venv\uctmaths_venv\lib\site-packages\django\core\mail\backends\smtp.py", line 99, in send_messages
    sent = self._send(message)
  File "C:\Users\dspies\work\uct\git\venv\uctmaths_venv\lib\site-packages\django\core\mail\backends\smtp.py", line 112, in _send
    for addr in email_message.recipients()]
  File "C:\Users\dspies\work\uct\git\venv\uctmaths_venv\lib\site-packages\django\core\mail\message.py", line 105, in sanitize_address
    nm, addr = addr
TypeError: 'NoneType' object is not iterable`

However, it disappeared after restarting the process and I can't get it to reappear. Can I ignore this

**Comment by yaseen-mowzer-hexagon** [comment]
That looks like it could be triggered by having an invalid email address in the database.

**Pull request merged**